src/MongoDB.Driver.Core/Core/Authentication/DefaultAuthenticator.cs

```diff
@@ -35,7 +35,7 @@ public class DefaultAuthenticator : IAuthenticator
         // fields
         private readonly UsernamePasswordCredential _credential;
         private readonly IRandomStringGenerator _randomStringGenerator;
-        private IAuthenticator _speculativeAuthenticator;
+        private readonly Lazy<IAuthenticator> _speculativeAuthenticator;
```

Comment:
The changes in this file are not super required, but I think now it looks a bit better